launch/docker.go

```diff
@@ -0,0 +1,85 @@
+package launch
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+)
+
+type docker struct {
+	volume            string
+	setupImage        string
+	setupImageVersion string
+}
+
+var _ runner = (*docker)(nil)
+var execCommand = exec.Command
+
+const (
+	artifactsDir = "artifacts"
+	logFile      = "builds.log"
+)
+
+func newDocker(setupImage, setupImageVer string) runner {
+	return &docker{
+		volume:            "SD_LAUNCH_BIN",
+		setupImage:        setupImage,
+		setupImageVersion: setupImageVer,
+	}
+}
+
+func (d *docker) setupBin() error {
+	err := execCommand("docker", "volume", "create", "--name", d.volume).Run()
+	if err != nil {
+		return fmt.Errorf("failed to create docker volume")
+	}
+
+	mount := fmt.Sprintf("%s:/opt/sd/", d.volume)
+	image := fmt.Sprintf("%s:%s", d.setupImage, d.setupImageVersion)
+	cmd := execCommand("docker", "container", "run", "--rm", "-v", mount, image, "--entrypoint", "/bin/echo set up bin")
```

Comment:
I got that error when I run sd-local. We need to add `local_run.sh` to the launcher image.
```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "exec: \"/opt/sd/local_run.sh\": stat /opt/sd/local_run.sh: no such file or directory": unknown.
2020/02/10 13:34:57 failed to run build: exit status 127
```

Reply:
I fixed it in https://github.com/screwdriver-cd/launcher/pull/331